File: java/jcf/Stack.java
package BuiltIn;

import java.util.Stack;

public class Stack {

    public static void main(String[] args) {
        Stack<Integer> stack = new Stack<>();

        // Push elements onto the stack
        stack.push(1);
        stack.push(2);
        stack.push(3);

        // Pop elements from the stack
        int topElement = stack.pop();
        System.out.println(topElement); // Output: 3

        // Peek at the top element
        int peekElement = stack.peek();
        System.out.println(peekElement); // Output: 2

        // Check if the stack is empty
        boolean isEmpty = stack.isEmpty();
        System.out.println(isEmpty); // Output: false

        // Get the size of the stack
        int size = stack.size();
        System.out.println(size); // Output: 2
    }
}

File: cpp/stl/stack.cpp
#include <iostream>
#include <stack>

int main() {
    // Create a stack of integers
    std::stack<int> myStack;

    // Push elements onto the stack
    myStack.push(10);
    myStack.push(20);
    myStack.push(30);

    // Print the top element of the stack
    std::cout << "Top element of the stack: " << myStack.top() << std::endl;

    // Pop elements from the stack
    myStack.pop();

    // Print the top element after popping
    std::cout << "Top element after popping: " << myStack.top() << std::endl;

    // Check if the stack is empty
    std::cout << "Is stack empty? " << (myStack.empty() ? "Yes" : "No") << std::endl;

    // Get the size of the stack
    std::cout << "Size of the stack: " << myStack.size() << std::endl;

    return 0;
}

